src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,441 @@
+        )
+
+    os.makedirs(run_dir, exist_ok=True)
+    logger.debug("Created directory for results at %s", run_dir)
+    return run_dir
+
+
+def print_results(result: EvaluationResult, results_files: List[str], model_name: str):
+    """
+    Prints a scoring report for DK-Bench
+    Args:
+        result (Evaluation):       ragas EvaluationResult to parse for scores
+                                   to each question.
+        results_files (List[str]): List of files with scores and responses
+        model_name (str):          Name of model that generated responses for
+                                   DK-Bench to evaluate againist reference answer.
+    Returns:
+        None
+    """
+    print("\n")
+    print("# DK-BENCH REPORT")
+    print(f"\n## MODEL: {model_name}\n")
+    total_score = 0
+    for i, score in enumerate(result.scores):
+        print(f"Question #{i+1}:     {score['domain_specific_rubrics']}/5")
+        total_score += score["domain_specific_rubrics"]
+
+    average = total_score / len(result.scores)
+    print("----------------------------")
+    print(f"Average Score:   {average:.2f}/5")
+    print(f"Total Score:     {total_score}/{len(result.scores)*5}\n")
+
+    print("Responses and scores written to:")
+    for file in results_files:
+        print(f"{file}")
+
+
+def create_excel_results_file(excel_file: str, result: EvaluationResult):
+    """
+    Writes an excel file based on the result of an evaluation.
+    The excel files has two sheets. The first is a summary sheet
+    with a table of individual question scores, average, total score and median.
+
+    The second sheet has the score, the question (user_input), reference,
+    response, model name, and evaluation run, similar to the contents of the
+    .jsonl and .csv files.
+
+    Args:
+        result (Evaluation):    ragas EvaluationResult to parse for scores
+                                for summary sheet and questions, references,
+                                and responses sheet.
+        excel_file (str):       Name of excel file to be created the summary
+                                and dataset sheets.
+    Returns:
+        None
+    """
+    summary_df = pd.DataFrame()
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    summary_df["scores"] = scores
+
+    question_indices = [f"Q{i + 1}" for i in range(len(summary_df))]
+    question_indices.append("Average")
+    question_indices.append("Total Score")
+    question_indices.append("Median")
+
+    average = summary_df.mean(axis=0, numeric_only=True)
+    total_score = summary_df.sum(axis=0, numeric_only=True)
+    median = summary_df.median(axis=0, numeric_only=True)
+
+    summary_df.loc[len(summary_df)] = average
+    summary_df.loc[len(summary_df)] = total_score
+    summary_df.loc[len(summary_df)] = median
+
+    summary_df.insert(0, "question index", question_indices)
+
+    response_df = result.dataset.to_pandas()
```

Comment:
What's the difference between `result.dataset.to_pandas()` and `result.to_pandas()`? I see that a few lines up you are copying over the scores into this `response_df`, but I think this information is already present in the base response from Ragas.